lib/puppet/network/http/api/indirected_routes.rb

```diff
@@ -190,14 +190,21 @@ def accepted_response_formatter_or_json_for(model_class, request)
 
   def read_body_into_model(model_class, request)
     data = request.body.to_s
-
-    format = request.format
-    begin
-      model_class.convert_from(format, data)
-    rescue => e
-      raise Puppet::Network::HTTP::Error::HTTPBadRequestError.new(
-        _("The request body is invalid: %{message}") % { message: e.message })
+    formatter = request.formatter
+
+    if formatter && formatter.supported?(model_class)
+      begin
+        return model_class.convert_from(formatter.name.to_s, data)
+      rescue => e
+        raise Puppet::Network::HTTP::Error::HTTPBadRequestError.new(
+          _("The request body is invalid: %{message}") % { message: e.message })
+      end
     end
+
+    #TRANSLATORS "mime-type" is a keyword and should not be translated
+    raise Puppet::Network::HTTP::Error::HTTPUnsupportedMediaTypeError.new(
```

Comment:
Oh look at that, this later commit responds with the 415 - disregard my earlier comment!